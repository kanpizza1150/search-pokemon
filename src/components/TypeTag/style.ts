import styled from 'styled-components'

enum ColorTags {
  Grass = 'green',
  Poison = 'purple',
  Fire = 'orange',
  Water = 'lightblue',
  Normal = 'grey',
  Bug = 'darkgreen',
  Ground = 'brown',
  Fairy = 'pink',
  Fighting = 'red',
  Rock = 'brown',
  Electric = 'yellow',
  Dragon = 'yellow',
  Psychic = 'yellow',
  Flying = 'yellow',
  Ice = 'yellow',
  Ghost = 'yellow',
  Dark = 'yellow',
  Steel = 'yellow',
}

interface Props {
  colorType: string
}

const getBGColor = (color: string): string =>
  `background-color:${ColorTags[color]};`

export const TagWrapper = styled.div`
  padding: 0.5rem;
  text-align: center;
  ${(props: Props) => getBGColor(props.colorType)};
  border-radius: 5px;
  margin: 0.5rem;
  font-size: 1rem;
`
