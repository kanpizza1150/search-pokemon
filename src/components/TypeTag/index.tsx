import React from 'react'
import { TagWrapper } from './style'

interface IProps {
  type: string
}
const TypeTag: React.FC<IProps> = ({ type }) => {
  return <TagWrapper colorType={type}>{type}</TagWrapper>
}
export default TypeTag
