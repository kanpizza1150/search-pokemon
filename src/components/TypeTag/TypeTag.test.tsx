import { render } from '@testing-library/react'
import TypeTag from './index'

describe('LoadingSpinner Component', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<TypeTag type='' />)
    expect(asFragment).toMatchSnapshot()
  })
  it('Should render text correctly', () => {
    const { getByTestId } = render(<TypeTag type='test' />)
    expect(getByTestId('tag-test').textContent).toBe('test')
  })
})
