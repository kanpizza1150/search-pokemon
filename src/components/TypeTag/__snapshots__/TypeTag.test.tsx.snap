// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner Component Should match snapshot 1`] = `[Function]`;

exports[`LoadingSpinner Component Should render text correctly 1`] = `[Function]`;
